### PR TITLE
Use fallback user avatar in cases where the user is unknown to us

### PR DESCRIPTION
**Pull request by Kicu**
### Details
 - this is PR is a followup after this: https://github.com/Expensify/App/pull/39229 was reverted, it also fixes issues that were found after the original PR merge
 - the logic that decides which avatar to show is now consolidated into the `<Avatar>` component, almost all calls to `getAvatar` and `getDefaultAvatar` were removed as they were mostly concerned with showing a correct SVG avatar version - now `<Avatar>` does all this
 - right now the source of truth is almost always `avatar` prop from `personalDetails`. It is returned from backend, and any user "known" to us will have it set
 - we will now display `FallbackAvatar` when:
   - there is no details/`accountID`
   - when `accountID` was optimistically set in TS code
   - as a final fallback in Avatar component (that behaviour is unchanged)
 - **special note for icons:** whenever we create icons array/object we have to explicitly set `FallbackAvatar` as source. Otherwise nothing would be shown because of patterns like this one: https://github.com/Expensify/App/blob/main/src/components/OptionRow.tsx#L209

#### Extra notes for reviewer:
 - please note that a PR affecting workspace avatars logic was merged some time ago, but after the revert: https://github.com/Expensify/App/pull/39637/files - it introduced usage of `accountID` for both user account and policy id.
 Because of this we were already passing accountID to Avatar component in some places
 - ☝️ introduced a bug related to Workspace avatar colors styling, if you notice something wrong with WS avatar please verify that its not already on main - this line seems problematic: https://github.com/Expensify/App/blob/main/src/components/Avatar.tsx#L97

CC @s77rt 

### Fixed Issues

$ https://github.com/Expensify/App/issues/38743
$ https://github.com/Expensify/App/issues/40989
$ https://github.com/Expensify/App/issues/40996

PROPOSAL:

### Tests

1. go to every place in the app where we can display an unknown user and verify that the avatar next to them is the fallback one (and not the colorful one)
2. places to verify:
 - search option from Chats list
 - searching users from "Start chat"
 - searching users from "Split expense"
 - searching users from "Pay someone"
 - searching users from "Assign Task"
 - searching users from Workspace list > Workspace > Members > Invite member 
3. test https://github.com/Expensify/App/issues/41014:
verify that small avatars in mentions from chat are correct including Concierge avatar
4. test https://github.com/Expensify/App/issues/40989:
 - go to Workspace list > Workspace > Members
 - turn on offline mode (disable network in devtools) and try adding a non-existing user to members
 - verify that profile page opens
5. test https://github.com/Expensify/App/issues/40996:
 - check if the avatar in quick action from Popover menu is the same as newly added user

### QA Steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
        - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/3929868/aebd1b4d-df48-40ef-933f-e2829bf1875f

</details>

<details>
<summary>Android: mWeb Chrome</summary>

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/3929868/419d996d-c16f-4b86-ac34-4cef462c5327

</details>

<details>
<summary>iOS: mWeb Safari</summary>

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/3929868/a42534b6-d59e-4af8-bd18-54da22085660

https://github.com/Expensify/App/assets/3929868/241d09d8-23d4-4ee6-9447-3793917779f6

https://github.com/Expensify/App/assets/3929868/39a43fe9-e9f0-4f26-9efb-ad4bbcd37608

</details>

<details>
<summary>MacOS: Desktop</summary>

</details>

### CODE REVIEW

File: src/components/Avatar.tsx
Comment (Kicu) [inline src/components/Avatar.tsx:85]: @s77rt any better ideas for `accountID as number`?

the logic here is
- if `isWorkspace` is true -> the `accountID` will be a `policyID` which is a `string`
- if `isWorkspace` false -> accountID is standard user account id which is a `number`

I dislike using casting via `as`, but not sure how to improve this line

File: src/pages/ProfilePage.tsx
Comment (s77rt) [inline src/pages/ProfilePage.tsx:1]: The changes here seems unrelated. Can you please double check

File: src/components/AvatarWithIndicator.tsx
Comment (s77rt) [inline src/components/AvatarWithIndicator.tsx:18]: Adding a comment to check https://github.com/Expensify/App/issues/40991 if it's something we can fix here

Comment (Kicu) [reply src/components/AvatarWithIndicator.tsx:18]: followed up with a comment and I will spend some time today digging into this bug

File: src/pages/ProfilePage.tsx
Comment (Kicu) [reply src/pages/ProfilePage.tsx:1]: You're asking about `ProfilePage`? So the changes are deliberate, and related to fixing https://github.com/Expensify/App/issues/40989.

Basically right now (on main) the decision to show most of the content is based on this line:
```js
const hasMinimumDetails = !isEmptyObject(details.avatar);
```
 (https://github.com/Expensify/App/blob/main/src/pages/ProfilePage.tsx#L154)

But we know that there will be **no** avatar in case we're offline and rely on optimistic data - as I removed the default avatar from optimistic data almost everywhere.
All the other basic data is there, so I just allow everything to display and the Fallback avatar will be correctly shown. 
I also disable clicking on the avatar when it's the fallback displayed

All the other changes are just indentation moved 1 level up. I tested this both in offline and online and it behaves correct I believe.

File: src/components/AvatarWithIndicator.tsx
Comment (s77rt) [inline src/components/AvatarWithIndicator.tsx:17]:
```suggestion
    accountID: number | undefined;
```

This should help catch cases where we are missing to pass accountID.

(do same with Avatar component)

**Review by s77rt — CHANGES_REQUESTED** [review]

File: src/types/onyx/OnyxCommon.ts
Comment (Kicu) [reply src/types/onyx/OnyxCommon.ts:31]: Because originally they were not optional, and this was changed only recently in this PR: https://github.com/Expensify/App/pull/39637/files#diff-3b661ee1b05411e77221e1333e48cf2ced284539b5b835070c5ec397606e21a2 - which is like a week old, introduced some regressions and I believe it was a bad idea. There was no real need to make them optional.

IMO Icons that have no source and no type are kinda meaningless. If every field in `Icon` is optional then `{}` could be a correct icon and that can lead to bugs when there is no source and no accountID.

Yeah I understand that we could drop the `?? FallbackAvatar` which is ugly, but in general I feel unsafe with the Icon type that could have everything optional. I'd rather keep the explicit `??` and be always forced to put something inside icon.

File: src/components/AvatarWithIndicator.tsx
Comment (Kicu) [reply src/components/AvatarWithIndicator.tsx:17]: Will do it for every Avatar.

What do you think about doing the same for `source`?
Now it looks kinda suspicious that I will have:`source?: string` but `accountID: number | undefined` almost as if accountID is more important than source.

**Comment by Kicu** [comment]
@s77rt thanks for all the comments, some notes from me:
 - please take a look at: https://github.com/Expensify/App/pull/41485/files as it introduces some changes related to workspace and it will most likely get merged before our PR
 - I decided that always passing explicitly `accountID={undefined}` looks out of place as its almost never done in this project, so I kept `accountID?` as optional
 - also: `<MenuItem>` renders `<Avatar>`, and `MenuItem` is rendered 100+ times in the App - I don't want to add `accountID` into all this places
 - about `Icon` read my comment, I would really like to return to source being not optional there I think it will be better for the codebase in the long run

File: src/libs/UserUtils.ts
Comment (Kicu) [reply src/libs/UserUtils.ts:84]: done

File: src/pages/home/report/ReportActionCompose/SuggestionMention.tsx
Comment (Kicu) [reply src/pages/home/report/ReportActionCompose/SuggestionMention.tsx]: done

File: src/pages/settings/InitialSettingsPage.tsx
Comment (Kicu) [reply src/pages/settings/InitialSettingsPage.tsx]: done in places where it made sense

File: src/components/Avatar.tsx
Comment (s77rt) [reply src/components/Avatar.tsx:85]: We can split the types into two and use union
```js
type AvatarProps = {
    ...
} & (
    | {
          type: typeof CONST.ICON_TYPE_AVATAR;
          accountID?: number;
      }
    | {
          type: typeof CONST.ICON_TYPE_WORKSPACE;
          accountID?: string;
      }
);
```

However for this to work we need to make the `type` prop required and remove its default value.

**Review by s77rt — CHANGES_REQUESTED** [review]

**Comment by s77rt** [comment]
Bug: In the task assignee the assign to me row has the fallback avatar

Screenshot 2024-05-11 at 2 16 16 PM

**Comment by Kicu** [comment]
All changes pushes @s77rt
I like the types suggestion but in the end I don't want to add any more changes because this PR is constantly growing, maybe in future we can add the union.


For the bug that you found there is something wrong in the whole chain of calls : `TaskAssigneeSelectorModalProps` -> `useOptionsList` -> `OptionsListUtils.createOptionList` -> `createOption` which returns a list of options that include empty Icon only for current user - so Im simply recreating the proper icon. Investigating this chain of calls was extremely difficult and in the end I don't know where to fix it. 

Screenshot 2024-05-13 at 16 51 06

**Comment by s77rt** [comment]
Can you please test if that bug is still reproducible? (without the workaround) I'm getting a different icon now which is also wrong but same as staging

Screenshot 2024-05-13 at 11 14 06 PM

**Comment by Kicu** [comment]
So without workaround it reproduces for me every time on web. I have a feeling that whether you get this bug is related to state of `personalDetails` key in onyx, because the logic that generates these options combines together a lot of data. If I explicitly search for myself then sometimes the avatar updates, but after a full page refresh I'm always back to the bug.

However I tried running it on ios and there I had the correct avatar :/
I think its better to keep the workaround if there are some cases where this bug appears

**Comment by s77rt** [comment]
In main / staging can you reproduce the bug? If not then it's a regression and we should fix it without workarounds

**Comment by s77rt** [comment]
Can you please resolve the conflicts and remove the workaround? I cannot reproduce this constantly but feels like we should handle it separately

**Comment by Kicu** [comment]
1 minute I pushed a bug

**Comment by Kicu** [comment]
Ok all done, workaround removed.

**Comment by Kicu** [comment]
omg I can't believe there was a test for this - I'm fixing the test now

**Comment by s77rt** [comment]
## Reviewer Checklist

- [X] I have verified the author checklist is complete (all boxes are checked off).
- [X] I verified the correct issue is linked in the `### Fixed Issues` section above
- [X] I verified testing steps are clear and they cover the changes made in this PR
    - [X] I verified the steps for local testing are in the `Tests` section
    - [X] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [X] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [X] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [X] I verified tests pass on **all platforms** & I tested again on:
    - [X] Android: Native
    - [X] Android: mWeb Chrome
    - [X] iOS: Native
    - [X] iOS: mWeb Safari
    - [X] MacOS: Chrome / Safari
    - [X] MacOS: Desktop
- [X] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [X] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [X] I verified that comments were added to code that is not self explanatory
    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [X] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [X] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [X] If a new component is created I verified that:
    - [X] A similar component doesn't exist in the codebase
    - [X] All props are defined accurately and each prop has a `/** comment above it */`
    - [X] The file is named correctly
    - [X] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [X] The only data being stored in the state is data necessary for rendering and nothing else
    - [X] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [X] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [X] All JSX used for rendering exists in the render method
    - [X] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [X] If any new file was added I verified that:
    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [X] If a new CSS style is added I verified that:
    - [X] A similar style doesn't already exist
    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [X] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [X] I verified that all the inputs inside a form are aligned with each other.
    - [X] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [X] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


android

</details>

<details>
<summary>Android: mWeb Chrome</summary>


Networking issue
</details>

<details>
<summary>iOS: Native</summary>


ios

</details>

<details>
<summary>iOS: mWeb Safari</summary>


mweb-safari

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>


web

</details>

<details>
<summary>MacOS: Desktop</summary>



https://github.com/Expensify/App/assets/16493223/a833aabd-002c-4983-9bd2-22e25f308a13


</details>

**Review by s77rt — APPROVED** [review]

**Comment by Kicu** [comment]
Also @grgia may be interested in taking a look at this since she has _some_ context about avatars I believe.

Also I think we should wait until this gets merged: https://github.com/Expensify/App/pull/41485 as the two will have a slight conflict and the other one is hotfixing a bug. Besides mine is bigger so it will take more time to merge the other way around.

So even if everything is good here please refrain from merging until 41485 lands on main

**Comment by Kicu** [comment]
Merged main and updated my code to work after this workspace avatars fix: https://github.com/Expensify/App/pull/41485/files

**Comment by s77rt** [comment]
@Kicu Can you please merge main again to fix the ts errors

**Review by grgia — DISMISSED** [review]
Code LGTM, Im going to spin a test build

**Comment by grgia** [comment]
@shawnborton or @dubielzyk-expensify I have one more avatar related PR- I just spun up a test build if you'd like to take a look. This one handles the changes to fallback avatars after the first take was reverted

**Comment by shawnborton** [comment]
Found a missing spot when viewing the workspace avatar from the workspace overview page:


https://github.com/Expensify/App/assets/2319350/4c7bb51e-c9b0-4fa6-a3a1-4952384aa0c9

**Comment by Kicu** [comment]
@shawnborton great find.

I checked that the same bug is currently appearing on web so its not introduced by my code, but rather another bug after this: https://github.com/Expensify/App/pull/39637

However if thats okey with you guys I could try to fix this within this PR as hopefully that will be faster.

**Comment by shawnborton** [comment]
Amazing, thanks!

File: src/components/AvatarWithIndicator.tsx
Comment (mountiny) [inline src/components/AvatarWithIndicator.tsx:16]: NAB
```suggestion
    /** Account id if it's user avatar */
```

File: src/components/Avatar.tsx
Comment (mountiny) [inline src/components/Avatar.tsx:84]:
```suggestion
    // If it's user avatar then accountID will be a number
```

File: src/libs/ReportUtils.ts
Comment (mountiny) [inline src/libs/ReportUtils.ts:6502]: This seems malformed to me, there should be the `*` at start of each line, I know it was wrong before, but lets fix it

File: src/libs/UserUtils.ts
Comment (mountiny) [inline src/libs/UserUtils.ts:82]:
```suggestion
```

**Review by mountiny — COMMENTED** [review]
@Kicu Couple minor comments, can you please sync up with main too? Thanks!

**Comment by Kicu** [comment]
@mountiny Done and done

**Review by mountiny — APPROVED** [review]
Thanks @Kicu Looks good to me

@grgia can you please review?

**Pull request merged**
